src/main/java/com/example/onlinebookstore/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.BookDtoWithoutCategoryIds;
+import com.example.onlinebookstore.dto.CategoryDto;
+import com.example.onlinebookstore.dto.CreateCategoryRequestDto;
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.BookMapper;
+import com.example.onlinebookstore.mapper.CategoryMapper;
+import com.example.onlinebookstore.model.Category;
+import com.example.onlinebookstore.repository.CategoryRepository;
+import com.example.onlinebookstore.repository.book.BookRepository;
+import com.example.onlinebookstore.service.CategoryService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    private final BookRepository bookRepository;
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+    private final BookMapper bookMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable).stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        Category category = categoryRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find category by id " + id));
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryDto save(CreateCategoryRequestDto requestDto) {
+        Category category = categoryMapper.toModel(requestDto);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public CategoryDto update(Long id, CreateCategoryRequestDto requestDto) {
+        Category category = categoryMapper.toModel(requestDto);
```

Comment:
```suggestion
        Category category = repository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException(
                        "No such a category with id: " + id
                ));
        mapper.updateCategoryFromDto(categoryDto, category);
        return mapper.toDto(repository.save(category));
```

Where 
`void updateCategoryFromDto(CategoryRequestDto categoryDto, @MappingTarget Category category);`